storagenode/pieces/readwrite.go

```diff
@@ -97,14 +275,25 @@ func (r *Reader) Read(data []byte) (int, error) {
 	return n, Error.Wrap(err)
 }
 
-// Seek seeks to the specified location.
+// Seek seeks to the specified location within the piece content (ignoring the header).
 func (r *Reader) Seek(offset int64, whence int) (int64, error) {
+	if whence == io.SeekStart && r.formatVersion >= storage.FormatV1 {
+		offset += V1PieceHeaderReservedArea
+	}
 	if whence == io.SeekStart && r.pos == offset {
 		return r.pos, nil
 	}
 
 	pos, err := r.blob.Seek(offset, whence)
 	r.pos = pos
+	if r.formatVersion >= storage.FormatV1 {
```

Comment:
Let's use a switch here rather than a check.

Reply:
My reluctance to use a switch here is because there may not be any change necessary here for future storage formats. It might well be `FormatV0` versus "everything else". If that ends up being the case, it may be unhelpful to have a switch requiring adding every new enumerator of FormatVersion to be added when they all just fall through to the same behavior.